pkgs/applications/video/obs-studio/default.nix

```diff
@@ -80,6 +81,10 @@ in mkDerivation rec {
         --prefix "LD_LIBRARY_PATH" : "${xorg.libX11.out}/lib:${vlc}/lib"
   '';
 
+  postFixup = stdenv.lib.optionalString stdenv.isLinux ''
+      addOpenGLRunpath $out/lib/lib*.so
```

Comment:
This probably needs a bit more investigation:
```
❯ ag -a dlopen result/lib
Binary file libobsglad.so matches.
Binary file libobsglad.so.0 matches.
Binary file libobs.so matches.
Binary file libobs.so.0 matches.
Binary file obs-plugins/linux-decklink.so matches.
Binary file obs-plugins/obs-ffmpeg.so matches.
Binary file obs-plugins/vlc-video.so matches.
```
Did you check `obs-ffmpeg.so` only `dlopen`s `libavcodec.so`, which should be fixed by https://github.com/NixOS/nixpkgs/pull/87588?

Reply:
I've just tested against the fixed ffmpeg from #87968 without the addOpenGLRunpath change in obs.
The ffmpeg change is not fixing it for obs, I get the following log output when starting obs:
`error: os_dlopen(libnvidia-encode.so.1->libnvidia-encode.so.1): libnvidia-encode.so.1: cannot open shared object file: No such file or directory`.

Reply:
So something inside `obs` tries to `dlopen` `libnvidia-encode.so.1` directly, okay.

I did a `ag -a libnv result/`:
```
Binary file lib/obs-plugins/obs-ffmpeg.so matches.
```

it looks like `lib/obs-plugins/obs-ffmpeg.so` is what's trying to open this - so we should need to patch this, which isn't covered by your glob `$out/lib/lib*.so`? :confused:

Reply:
I've added the call for the obs-plugins folder.

Reply:
I'm not sure if the other line is still necessary as well, but it's probably fine.